include/zephyr/sys_clock.h

```diff
@@ -146,11 +146,14 @@ typedef struct {
 
 /** @endcond */
 
-#if defined(CONFIG_SYS_CLOCK_EXISTS) && \
-	(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC == 0)
+#if defined(CONFIG_SYS_CLOCK_EXISTS) && !defined(CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME)
+
+#if (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC == 0)
 #error "SYS_CLOCK_HW_CYCLES_PER_SEC must be non-zero!"
```

Comment:
If you're mucking with the logic here anyway, BUILD_ASSERT() is the more modern way to do this sort of thing.

Reply:
updated